### PR TITLE
protect kube control plane operator instances

### PR DESCRIPTION
**Pull request by deads2k**
These resources are special because if there isn't an active operator managing them, the configuration can rot (certs for instance) and your cluster can die beyond a point where it is easy to recover.

/assign @sanchezl

### CODE REVIEW

File: pkg/admission/customresourcevalidation/config/deny_delete_cluster_config_resource.go [hunk 1]
```diff
@@ -31,23 +31,39 @@ func (p *admissionPlugin) Validate(attributes admission.Attributes) error {
...
+				return nil
+			}
+
+		default: // all other resources in operator.openshift.io can be deleted.
 			return nil
 		}
+	default: // for all other groups, do nothing
+		return nil
 	}
 	return admission.NewForbidden(attributes, fmt.Errorf("deleting required %s.%s resource, named %s, is not allowed", attributes.GetResource().Resource, attributes.GetResource().Group, attributes.GetName()))
```
Comment (smarterclayton) [inline pkg/admission/customresourcevalidation/config/deny_delete_cluster_config_resource.go]: oh my god so many commas.  remove the `, named %s,`

Comment (smarterclayton) [reply pkg/admission/customresourcevalidation/config/deny_delete_cluster_config_resource.go]: The commas on both side, that is

**Comment by enj** [comment]
@sferich888 will be happy :)

**Comment by deads2k** [comment]
> @sferich888 will be happy :)

We'll see.  This was to prove I could, but I'd still like to see someone, *anyone* accidentally type `oc delete kubeapiserver.operator.openshift.io --all --as=system:admin` during the beta.  At a certain point you have to say, "you know what, you don't just accidentally type all that out and if you delete all the machines, you end up in a similarly screwed state".

**Comment by sferich888** [comment]
> > @sferich888 will be happy :)

Well see, I am worried that this will keep us from being able to delete things (as a mode of correcting a problem).  I would like to see this enabled or disabled with some form of a feature gate (something we can turn on or off). IE: Safety on by default; but provide some form of a safety selector. 

> We'll see. This was to prove I could, but I'd still like to see someone, _anyone_ accidentally type `oc delete kubeapiserver.operator.openshift.io --all --as=system:admin` during the beta. At a certain point you have to say, "you know what, you don't just accidentally type all that out and if you delete all the machines, you end up in a similarly screwed state".

I don't see this "kill" a cluster (when I run this command). As with https://bugzilla.redhat.com/show_bug.cgi?id=1684652#c10 you can recover from this; by running: 

`$ oc delete $(oc get pods -n openshift-kube-apiserver-operator -o name) -n openshift-kube-apiserver-operator`

`$ oc logs kube-apiserver-operator-586d9c8944-dsxhw -n openshift-kube-apiserver-operator`
> 51s         Warning   StatusNotFound               deployment/kube-apiserver-operator   Unable to determine current operator status for kube-apiserver

You can't find the operator (but I feel that's expected). 

`$ for pod in $(oc get pods -n openshift-kube-apiserver-operator -o name); do oc logs $pod; done`
> Error from server (NotFound): pods "kube-apiserver-operator-586d9c8944-l9kjc" not found

In short, my updates in https://bugzilla.redhat.com/show_bug.cgi?id=1684652#c10 still seem to apply to this.

**Comment by deads2k** [comment]
we recover these now I think
